include/serialport.h

```diff
@@ -143,6 +143,19 @@ class MyFifo {
 	size_t used = 0;
 };
 
+enum SerialTypesE { // Also change src/dos/program_serial.cpp if you change this.
+	SERIAL_TYPE_DISABLED = 0,
+	SERIAL_TYPE_DUMMY,
+#ifdef C_DIRECTSERIAL
+	SERIAL_TYPE_DIRECT_SERIAL,
+#endif
+#if C_MODEM
+	SERIAL_TYPE_MODEM,
+	SERIAL_TYPE_NULL_MODEM,
+#endif
+	SERIAL_TYPE_COUNT
+};
```

Comment:
Consider letting enums be fully populated, because they're just IDs without dependencies into the OS or libraries. 
The benefit is that the IDs won't "move around" based on compile features. 
For example, as currently stated, if `C_DIRECTSERIAL` is disabled then `SERIAL_TYPE_MODEM` is given the ID of 2. 
But if  `C_DIRECTSERIAL` is enabled then `SERIAL_TYPE_DIRECT_SERIAL` has the ID of 2.

If we eliminate the preprocessor controls, then SERIAL_TYPE_DIRECT_SERIAL is always 2 and SERIAL_TYPE_DIRECT_SERIAL is always 3 (no matter how the program's configured).